api/auth.go

```diff
@@ -0,0 +1,28 @@
+package api
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+)
+
+func authorized(authSecret string, next http.Handler) http.HandlerFunc {
+	return func(rw http.ResponseWriter, r *http.Request) {
+		authHeader := r.Header.Get("Authorization")
```

Comment:
This is something I hadn't actually considered - should we be auth-ing in our callbacks? We're currently just doing a raw POST to whatever URL you pass us

Reply:
Especially since we'll be exposing the `task-runner` API to internet, I think we _should_ have this auth here as well!

If it is too much of a lift to do it on the `catalyst-api` side, I could make an "internal" authorization logic here, like putting the auth secret in the query string. That would be a little less secure overall tho (any service logging the URLs will log the secret too), but would work. So LMK if you think you'll be able to add the secret on the callback or if I should do something like that